framework/src/headers/icmp.rs

```diff
@@ -0,0 +1,50 @@
+use headers::ip::IpHeader;
+use std::default::Default;
+use std::fmt;
+
+pub struct IcmpV6Header<T> {
+    msg_type: u8,
+    code: u8,
+    checksum: u16
+}
+
+pub const ROUTER_ADVERTISEMENT: int = 134;
```

Comment:
as part of the functional test, we should work on how we become selective on which we accept and which we just want to forward through. Maybe an enum?